src/o365/spo/commands/listitem/listitem-batch-add.ts

```diff
@@ -0,0 +1,470 @@
+import GlobalOptions from '../../../../GlobalOptions';
+import request from '../../../../request';
+import {
+  CommandOption,
+  CommandValidate
+} from '../../../../Command';
+import SpoCommand from '../../../base/SpoCommand';
+import Utils from '../../../../Utils';
+//import { ListItemInstance } from './ListItemInstance';
+import { FolderExtensions } from '../../FolderExtensions';
+import * as path from 'path';
+import { Transform } from 'stream';
+const vorpal: Vorpal = require('../../../../vorpal-init');
+const csv = require('@fast-csv/parse');
+import { createReadStream } from 'fs';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  listId?: string;
+  listTitle?: string;
+  contentType?: string;
+  folder?: string;
+  path: string;
+  batchSize:number;
+}
+
+class SpoListItemAddCommand extends SpoCommand {
+
+  public allowUnknownOptions(): boolean | undefined {
+    return false;
+  }
+
+  public get name(): string {
+    return commands.LISTITEM_BATCH_ADD;
+  }
+
+  public get description(): string {
+    return 'Creates a list item in the specified list for each row in the specified .csv file';
+  }
+
+  public getTelemetryProperties(args: CommandArgs): any {
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.listId = typeof args.options.listId !== 'undefined';
+    telemetryProps.listTitle = typeof args.options.listTitle !== 'undefined';
+    telemetryProps.contentType = typeof args.options.contentType !== 'undefined';
+    telemetryProps.folder = typeof args.options.folder !== 'undefined';
+    return telemetryProps;
+  }
+
+  public commandAction(cmd: CommandInstance, args: CommandArgs, cb: (err?: any) => void): void {
+    let lineNumber: number = 0;
+    let contentTypeName: string | null = null;
+    let listRestUrl: string | null = null;
+    let maxBytesInBatch: number = 1000000; // max is  1048576
+    let rowsInBatch: number = 0;
+    let batchCounter = 0;
+      let recordsToAdd = "";
+    let csvHeaders: Array<string>;
+    const generateUUID = function () {
+      var d = new Date().getTime();
+      var uuid = 'xxxxxxxx-xxxx-4xxx-yxxx-xxxxxxxxxxxx'.replace(/[xy]/g, function (c) {
+        var r = (d + Math.random() * 16) % 16 | 0;
+        d = Math.floor(d / 16);
+        return (c == 'x' ? r : (r & 0x7 | 0x8)).toString(16);
+      });
+      return uuid;
+    }
+    const parseResults = (response: any, cmd: CommandInstance, cb: (err?: any) => void): void => {
+      let responseLines: Array<string> = response.toString().split('\n');
+      // read each line until you find JSON... 
+      for (let responseLine of responseLines) {
+        try {
+          //check for error 
+
+          if (responseLine.startsWith("HTTP/1.1 5")) { //any 500 errors (like timeout), just stop
+            cmd.log("An HTTP 5xx error was returned from SharePoint. Please retry with a lower --batchsize ")
+            cb(responseLine);
```

Comment:
Calling `cb` here won't stop the command from executing

Reply:
calling cb with a CommandError now. that stops it.